Tema-Java8/src/main/java/java8/Java8Main.java

```diff
@@ -0,0 +1,278 @@
+    public static Set<String> randomString = new HashSet<>();
+
+    public static void main(String[] args) {
+
+        // 1) Create and Add 5 different instances of an object in a list.
+        List<Flower> flowerList = Arrays.asList(
+                new Flower("Magnolia", "white", 6, 45, 30),
+                new Flower("Lilac", "purple", 72, 12, 40),
+                new Flower("Peony", "pink", 13, 60, 20),
+                new Flower("Sunflower", "yellow", 54, 200, 90),
+                new Flower("Rose", "red", 45, 75, 60)
+        );
+
+
+        System.out.println("The list of flowers which name starts with 'M' and contains letter 'a' is: "
+                + startsWithMContainsA(flowerList, "M", "a") + "\n");
+
+        System.out.println("The flower with the least living time is: "
+                + minimumLifeTime(flowerList) + "\n");
+
+        System.out.println("The 'Max' String of the 5 generated is: "
+                + maxOfRandomStrings(5) + "\n");
+
+        System.out.println("The map of random integers and their last 2 numbers: "
+                + mapOfRandomIntegersAndTheirSquaredNumber(5) + "\n");
+
+        System.out.println("How many values of the map are greater than 10: "
+                + howManyAreGreaterThanTen() + "\n");
+
+
+        sortRandomStringsSetInReverseOrder(randomString);
+
+        System.out.println(sortFlowersByHeight(flowerList) + "\n");
+
+
+        System.out.println("How many flowers from the list have even number of petals: "
+                + countOfEvenNumberOfPetals(flowerList) + "\n");
+
+        System.out.println("The shortest String from the set is: " + shortestString() + "\n");
+
+        System.out.println("And last, but not least: *THE OPTIONAL*: ");
+        optionalExample();
+    }
+
+    /**
+     * 11)
+     * I needed to create another set because all the random ones are length 10. For some reason it printed me one of
+     * them, even though they have equal length.
+     * <p>
+     * I think the easiest, maybe fastest way to find the shortest string by length is by using the min method that use
+     * an Comparator. I think it takes the shortest code to resolve and it's the first idea that popped in mind.
+     * Plus, all I needed was to create the set and the return, the rest is from java.util
+     * <p>
+     * If they have the same length the result is the "min" one after comparing character with character
+     *
+     * @return the shortest string in the set
+     */
+    private static String shortestString() {
+        Set<String> newSetToCompareByLength = new HashSet<>();
+        newSetToCompareByLength.add("a2");
+        newSetToCompareByLength.add("a1");
+        newSetToCompareByLength.add("abc");
+        newSetToCompareByLength.add("abcd");
+        newSetToCompareByLength.add("abcde");
+
+        return newSetToCompareByLength.stream()
+                .min(Comparator.comparing(String::length))
```

Comment:
it is fast, but it's not always the fastest, check .reduce for multiple cases when you consider